src/Tools/dotnet-monitor/Egress/EgressService.cs

```diff
@@ -137,26 +159,26 @@ private static EgressArtifactSettings CreateSettings(IEndpointInfo source, strin
 
         private void Reload()
         {
-            _providerTypeMap.Clear();
-            _optionsTypeMap.Clear();
+            _egressProviderMap.Clear();
 
             // Deliberately fill the maps in the reverse order of how they are accessed
             // by the GetProvider method to avoid indeterminate accessing of the option
             // information.
             foreach (IEgressProviderConfigurationProvider provider in _providers)
```

Comment:
If we change the `IEgressProviderConfigurationProvider.ProviderType` property to `IEnumerable<string> ProviderTypes { get; }`, then the service configuration doesn't need to iterate the configuration settings. The `EgressService` then just needs to iterate the ProviderTypes in an extra loop and the remainder largely remains unchanged.

Then we can change the `Configuration` property into something like `IConfiguration GetConfiguration(string)` method that takes a provider name (not the provider type) and returns the associated configuration. This will allow the `EgressProviderConfigureNamedOptions<T>` to bind the options to the correct configuration section.

Finally, `EgressProviderConfigurationChangeTokenSource<T>` can be updated to build a [CompositeChangeToken](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.primitives.compositechangetoken?view=dotnet-plat-ext-6.0) from the iterative calls to `IEgressProviderConfigurationProvider.GetConfiguration`.

With those changes, I think everything higher in the stack (other than passing the provider name through IEgressProvider) won't be necessary.